# Ellestad_etal_2025_APPS_Hybridizations
This repository contains all the scripts and data associated with the following publication: DOI: __


# Description
Phylogenomic studies increasingly reveal the pervasiveness of conflicting evolutionary histories and phylogenetic discordance. These conflicts arise from biological processes such as gene flow, incomplete lineage sorting (ILS), gene duplication and loss often the result of whole genome duplication events, or by methodological noise in data assembly and filtering. Among these, hybridization and introgression may be particularly influential in shaping phylogenetic discordance and offer valuable insights into evolutionary processes. These processes have long been recognized as fundamental to evolution, but detecting and disentangling their significance, especially across different evolutionary time scales, has proven challenging. Recent advances in phylogenetic network methods now allow simultaneous estimation of ILS and hybridization events, thereby improving our ability to detect and quantify these common plant evolutionary processes and patterns. 

Given the complexity of these bioinformatic approaches, the following tutorial was developed to guide researchers through these methodological processes, making it applicable to other study systems. This tutorial serves as a template for researchers aiming to investigate the complex evolutionary processes that influence the diversity of life. Researchers may contact moore.erika.r@gmail.com or pllestad@memphis.edu regarding questions in methodology. This pipeline implements code in multiple languages, mainly bash, R, python, and julia, and often uses conda as a way to quickly and cleanly install packages on a HPC.

Access the tutorial website [here](https://erika-r-moore.github.io/Ellestad_etal_2025_APPS_Hybridizations/). Corresponding input and output files and images can be found in the Example_files folder above. Please reference the README.md in Example_files for file type information.



# Data Availability
Raw sequence data are available through the National Center for Biotechnology Information (NCBI) Sequent Read Archive (SRA) and GeneBank portals. Previously published transcriptome data are in BioProjects [PRJNA994483](https://www.ncbi.nlm.nih.gov/bioproject/PRJNA994483), [PRJNA636629](https://www.ncbi.nlm.nih.gov/bioproject/?term=PRJNA636629), and [PRJEB4922](https://www.ncbi.nlm.nih.gov/bioproject/?term=PRJEB4922). Newly sequenced transcriptomes are deposited at NCBI SRA under BioProject [PRJNA1246973](https://www.ncbi.nlm.nih.gov/bioproject/PRJNA1246973/). The two newly sequenced Hyb-Seq data are uploaded to NCBI SRA under BioProject [PRJNA1246973](https://www.ncbi.nlm.nih.gov/bioproject/PRJNA1246973/), with the remaining taxa found in BioProject [PRJNA540287](https://www.ncbi.nlm.nih.gov/bioproject/?term=PRJNA540287). Complete plastomes are uploaded to NCBI, under BioProject [PRJNA1246973](https://www.ncbi.nlm.nih.gov/bioproject/PRJNA1246973/).


# Caveats
All scripts are provided as is and are not guaranteed to work. If you have any issues with the scripts or need clarification, don't hesitate to reach out via email (moore.erika.r@gmail.com or pllestad@memphis.edu) or submit a GitHub issue. We are happy to help!


# Citation
If you use the scripts from this tutorial, please reference the associated publication: ____
